### PR TITLE
[Gluten-core] Support sha1/crc32 function using velox backend

### PR DESCRIPTION
**Pull request by izchen**
## What changes were proposed in this pull request?

Support sha1 function using velox backend.

Depend on Velox function support: https://github.com/oap-project/velox/pull/132

## How was this patch tested?

Added a ut case

### CODE REVIEW

File: backends-velox/src/test/scala/io/glutenproject/execution/VeloxStringFunctionsSuite.scala [hunk 1]
```diff
@@ -154,6 +154,13 @@ class VeloxStringFunctionsSuite extends WholeStageTransformerSuite {
       s"from lineitem limit $LENGTH") { checkOperatorMatch[ProjectExecTransformer] }
   }
 
+  test("sha1") {
+    runQueryAndCompare(s"select l_orderkey, sha1(l_comment) " +
+      s"from lineitem limit $LENGTH") { checkOperatorMatch[ProjectExecTransformer] }
```
Comment (PHILO-HE) [inline backends-velox/src/test/scala/io/glutenproject/execution/VeloxStringFunctionsSuite.scala:159]: It looks it execution goes into fallback path due to validation failed, which makes the test fail:
```
Validation failed for expression in ProjectRel due to:Found incompatible return types for 'sha1' (VARBINARY vs. VARCHAR) for input types (VARBINARY).
```

File: backends-velox/src/test/scala/io/glutenproject/execution/VeloxStringFunctionsSuite.scala [hunk 2]
```diff
@@ -154,6 +154,13 @@ class VeloxStringFunctionsSuite extends WholeStageTransformerSuite {
...
+    runQueryAndCompare(s"select l_orderkey, sha1(null) " +
```
[the remark this answers is not in the document]
Comment (izchen) [reply backends-velox/src/test/scala/io/glutenproject/execution/VeloxStringFunctionsSuite.scala:160]: You're right, I'll add this conf

File: backends-velox/src/test/scala/io/glutenproject/execution/VeloxStringFunctionsSuite.scala (hunk 1 above)
Comment (izchen) [reply backends-velox/src/test/scala/io/glutenproject/execution/VeloxStringFunctionsSuite.scala:159]: Failure is due to the need to merge  https://github.com/oap-project/velox/pull/132 before this.

File: backends-velox/src/test/scala/io/glutenproject/execution/VeloxStringFunctionsSuite.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (izchen) [reply backends-velox/src/test/scala/io/glutenproject/execution/VeloxStringFunctionsSuite.scala:163]: done

File: backends-velox/src/test/scala/io/glutenproject/execution/VeloxStringFunctionsSuite.scala (hunk 1 above)
Comment (izchen) [reply backends-velox/src/test/scala/io/glutenproject/execution/VeloxStringFunctionsSuite.scala:162]: https://github.com/oap-project/velox/pull/132 has been merged

**Comment by izchen** [comment]
It looks like the implicitly added `cast(null:NullType as binary)` is failing. I will look at how to fix this

**Comment by PHILO-HE** [comment]
A CI checking for CH backend failed. It should be irrelevant to your PR. Please rebase your code.

**Comment by PHILO-HE** [comment]
The CI failure is caused by a commit in main branch. Let's wait for the fix and then re-run the CI jobs.

**Review by PHILO-HE — APPROVED** [review]
The patch looks good to me. Will be merged soon.

**Comment by izchen** [comment]
Thanks @PHILO-HE !

**Pull request merged**